libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -229,7 +229,14 @@ void AP_AHRS::add_trim(float roll_in_radians, float pitch_in_radians, bool save_
 // Set the board mounting orientation, may be called while disarmed
 void AP_AHRS::update_orientation()
 {
-    const enum Rotation orientation = (enum Rotation)_board_orientation.get();
+    const Rotation orientation = (enum Rotation)_board_orientation.get();
+    if (last_orientation == orientation) {
```

Comment:
This breaks any changes to the orientation on the custom orientation, and would need to be done inside the first if block on line 240.